### PR TITLE
Switch pytest integration to hook

### LINKED ISSUE #152: Pytest 8.1.0 removed _pytest.runner.call_runtest_hook
Newest pytest removed the `_pytest.runner.call_runtest_hook`, which flexmock depends on.

```
*****************
Running unittest
*****************
PYTHONPATH=/Users/tvuotila/flexmock python3 tests/test_unittest.py
Traceback (most recent call last):
  File "/Users/tvuotila/flexmock/tests/test_unittest.py", line 6, in <module>
    from flexmock import flexmock
  File "/Users/tvuotila/Library/Caches/pypoetry/virtualenvs/flexmock-RTiH4lMC-py3.12/lib/python3.12/site-packages/flexmock/__init__.py", line 2, in <module>
    from flexmock import _integrations  # pylint: disable=unused-import
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tvuotila/Library/Caches/pypoetry/virtualenvs/flexmock-RTiH4lMC-py3.12/lib/python3.12/site-packages/flexmock/_integrations.py", line 101, in <module>
    saved_pytest = runner.call_runtest_hook
                   ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module '_pytest.runner' has no attribute 'call_runtest_hook'
make: *** [unittest] Error 1
```

Reproduction:
```
pip install pytest==8.1.0
make test
```

Note: pytest >7.4.2 depends on Python >=3.8 and flexmock supported Python version range is ^3.7.1. Thus, `poetry update pytest` won't install the newest pytest.

### PR DESCRIPTION
**Pull request by mfocko**
Switch the pytest integration to the hook instead of wrapping the runner manually.

Fixes #152

TODO:
- [x] Adjust specfile for Fedora, cause apparently the pytest module is not included.
  - Decided to use the specfile directly from the Fedora, since the one, here, is not being kept up to date.
- [x] Rebase
- [x] Test against 8.1.1
- [x] Changelog
- [x] Docstrings

### CODE REVIEW

**Comment by ollipa** [comment]
Thank you for the fix! The tests seem to fail for Pytest 5 but I think we should drop support for Pytest 5 in the next release along with Python 3.7.

File: src/flexmock/_pytest_plugin.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Hook flexmock into Pytest.
```
Comment (ollipa) [inline src/flexmock/_pytest_plugin.py]: Can you change this comment to a docstring?

File: src/flexmock/_pytest_plugin.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+from _pytest.runner import CallInfo, ExceptionInfo, Item, TestReport
+
+from flexmock._api import flexmock_teardown
+
+
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_makereport(
+    item: Item,  # pylint: disable=unused-argument
+    call: CallInfo[None],
+) -> Generator[None, None, None]:
```
Comment (ollipa) [inline src/flexmock/_pytest_plugin.py:18]: Can you add a function docstring here?

**Review by ollipa — APPROVED** [review]
Looks good and test suite passes! Can you update the CHANGELOG.md?

The linting should pass after you add docstring to the new pytest plugin file.

I made a PR to drop support for Python 3.7 and Pytest 5.x (https://github.com/flexmock/flexmock/pull/154). Feel free to rebase your changes onto that.

It seems that pytest yanked version 8.1.0 so anyone testing this branch should install pytest 8.1.0 explicitly.

File: src/flexmock/_pytest_plugin.py (hunk 2 above)
Comment (mfocko) [reply src/flexmock/_pytest_plugin.py:18]: added

File: src/flexmock/_pytest_plugin.py (hunk 1 above)
Comment (mfocko) [reply src/flexmock/_pytest_plugin.py]: done

**Comment by mfocko** [comment]
Sorry about the trouble with the pylint, I'm getting kinda inconsistent results with the `make` command :eyes: locally it fails on mypy already (untyped decorator), also pylint produces way more warnings (mostly newer type aliases for union/optional/etc.)

**Comment by ollipa** [comment]
> Sorry about the trouble with the pylint, I'm getting kinda inconsistent results with the `make` command 👀 locally it fails on mypy already (untyped decorator), also pylint produces way more warnings (mostly newer type aliases for union/optional/etc.)

I think it is just because you are using newer Python version locally. The CI uses the minimum supported Python version.

Thank your work on this. I'll merge this PR soon and create a new release.

**Pull request merged**